aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -153,49 +153,71 @@ const KernelFunction& OperatorEntry::computeDispatchTableEntry(const c10::Dispat
   return computeDispatchTableEntryWithDebug(dispatcher, dispatch_key).first.kernel;
 }
 
+bool OperatorEntry::hasKernelForDispatchKey(DispatchKey dispatch_key) const {
+  auto kern_it = kernels_.find(dispatch_key);
+  return kern_it != kernels_.end();
+}
+
+const AnnotatedKernel& OperatorEntry::getKernelForDispatchKey(DispatchKey dispatch_key) const{
+  TORCH_INTERNAL_ASSERT(hasKernelForDispatchKey(dispatch_key));
+  TORCH_INTERNAL_ASSERT(!kernels_.at(dispatch_key).empty());
+  TORCH_INTERNAL_ASSERT(kernels_.at(dispatch_key).front().kernel.isValid());
+  return kernels_.at(dispatch_key).front();
+}
+
 std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTableEntryWithDebug(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) const {
   auto dispatch_ix = static_cast<uint8_t>(dispatch_key);
   // [Note] DispatchTable computation
   // dispatchTable contains entries for runtime dispatch keys.
   // For any dispatch key, it'll pick a kernel using the following order:
   //  (1) Use kernel if it's directly registered to this key
   //  (2) Handle runtime keys that have kernels available from alias keys
-  //    (2.1) Use kernel from DispatchKey::Autograd if available
-  //    (2.2) For autograd backend keys, we use kernel from alias Math key (catchAll will be moved to Math)
+  //    (2.1) Use kernel from DispatchKey::Math if available. For autograd keys, we only use kernel from Math
+  //          when there's no direct registration to its corresponding backend key.
```

Comment:
Hey, sorry I didn't realize this during the earlier discussion but don't we have the same issue here regarding keys like AutogradOther that represent multiple backends? If any of them register a kernel, then registering a Math kernel to (e.g.) AutogradOther will cause it to be skipped, right?